### PR TITLE
adding create and switch schema

### PR DESCRIPTION
**Pull request by hichem-fazai**
This PR is to add : 
SchemaUtils.createSchema(schema) = > creates schema
SchemaUtils.dropSchema(schema) = > drops schema
SchemaUtils.setSchema(schema) = > sets the default schema to [schema]

### CODE REVIEW

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Schema.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        val createSchemaDDL = buildString {
+            append("CREATE SCHEMA ")
+            if (currentDialect.supportsIfNotExists) {
+                append("IF NOT EXISTS ")
+            }
+            append(identifier)
+
+            if(TransactionManager.current().connection.metadata { supportsSchemasInDataManipulation }
+                    || !TransactionManager.current().connection.metadata { supportsCatalogsInDataManipulation }) {
```
Comment (Tapac) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Schema.kt]:
```suggestion
            if(TransactionManager.current().connection.metadata { supportsSchemasInDataManipulation || !supportsCatalogsInDataManipulation }) {
```

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt [hunk 2]
```diff
@@ -288,4 +288,51 @@ object SchemaUtils {
...
+     *
+     * @sample org.jetbrains.exposed.sql.tests.shared.SchemaTests
+     */
+    fun setSchema(schema: Schema) {
+        with(TransactionManager.current()) {
+            val createStatements = schema.setSchemaStatement()
+
+            execStatements(false, createStatements)
+
+            if(currentDialect is MysqlDialect) {
```
Comment (Tapac) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt]: Could you comment on why we need to set a catalog for MySQL and not set a schema for other databases?

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/Default.kt [hunk 3]
```diff
@@ -478,6 +478,8 @@ interface DatabaseDialect {
     /** Returns `true` if the dialect supports the creation of sequences, `false` otherwise. */
     val supportsCreateSequence: Boolean get() = true
     /** Returns `true` if the dialect requires the use of a sequence to create an auto-increment column, `false` otherwise. */
+    val supportsCreateSchema: Boolean get() = TransactionManager.current().connection.metadata { supportsSchemasInDataManipulation }
+            || TransactionManager.current().connection.metadata { supportsCatalogsInDataManipulation }
```
Comment (Tapac) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/Default.kt]:
```suggestion
    val supportsCreateSchema: Boolean get() = TransactionManager.current().connection.metadata { supportsSchemasInDataManipulation || supportsCatalogsInDataManipulation }
```

File: exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcConnectionImpl.kt [hunk 4]
```diff
@@ -15,8 +15,11 @@ import java.sql.PreparedStatement
...
 
     // Oracle driver could throw excpection on catalog
-    override val catalog: String by lazy {
-        try { connection.catalog } catch (_: Exception) { null } ?: connection.metaData.userName ?: ""
+    override val catalog: String
+    get() =  try { connection.catalog } catch (_: Exception) { null } ?: connection.metaData.userName ?: ""
+
+    override val schema: String by lazy {
+        connection.schema
     }
```
Comment (Tapac) [inline exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcConnectionImpl.kt]: Afaik, some drivers (e.g. MySQL) could throw `AbstractMethodException` on an attempt to get `schema` from a connection. Also, I think it should be getter to as you add the possibility to change a schema.

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt [hunk 5]
```diff
@@ -288,4 +288,51 @@ object SchemaUtils {
             currentDialect.resetCaches()
         }
     }
+
+    /**
+     * Sets the current default schema to [schema]
```
Comment (Tapac) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt]: Please improve the comment to state what schema will be changed only for a current connection.

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt (hunk 2 above)
Comment (hichem-fazai) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt]: This is a workaround for mysql jdbc version lower or equal to 5.1.48. 
In this version, the `connection.catalog` doesn't change after executing `USE db` statement. So in this workaround I'm setting it manually.

For other databases, executing the command that changes the schema sets automatically the connection.schema field to the name of the schema we are switching to. So `connection.setSchema` will be useless.

I tried with Mysql jdbc version 6 and higher and it works fine. After  executing `USE db` statement, the `connection.catalog` returns the new schema name  without setting it manually. Also for MariaDB(which supports only catalogs just like mysql) , the problem doesn't appear.

**Pull request merged**